src/azure-cli/azure/cli/command_modules/acs/addonconfiguration.py

```diff
@@ -0,0 +1,656 @@
+    if workspace_resource_id.endswith('/'):
+        workspace_resource_id = workspace_resource_id.rstrip('/')
+    return workspace_resource_id
+
+
+def ensure_container_insights_for_monitoring(cmd,
+                                             addon,
+                                             cluster_subscription,
+                                             cluster_resource_group_name,
+                                             cluster_name,
+                                             cluster_region,
+                                             remove_monitoring=False,
+                                             aad_route=False,
+                                             create_dcr=False,
+                                             create_dcra=False):
+    """
+    Either adds the ContainerInsights solution to a LA Workspace OR sets up a DCR (Data Collection Rule) and DCRA
+    (Data Collection Rule Association). Both let the monitoring addon send data to a Log Analytics Workspace.
+
+    Set aad_route == True to set up the DCR data route. Otherwise the solution route will be used. Create_dcr and
+    create_dcra have no effect if aad_route == False.
+
+    Set remove_monitoring to True and create_dcra to True to remove the DCRA from a cluster. The association makes
+    it very hard to delete either the DCR or cluster. (It is not obvious how to even navigate to the association from
+    the portal, and it prevents the cluster and DCR from being deleted individually).
+    """
+    if not addon.enabled:
+        return None
+
+    # workaround for this addon key which has been seen lowercased in the wild
+    for key in list(addon.config):
+        if key.lower() == CONST_MONITORING_LOG_ANALYTICS_WORKSPACE_RESOURCE_ID.lower() and key != CONST_MONITORING_LOG_ANALYTICS_WORKSPACE_RESOURCE_ID:
+            addon.config[CONST_MONITORING_LOG_ANALYTICS_WORKSPACE_RESOURCE_ID] = addon.config.pop(
+                key)
+
+    workspace_resource_id = addon.config[CONST_MONITORING_LOG_ANALYTICS_WORKSPACE_RESOURCE_ID]
+    workspace_resource_id = sanitize_loganalytics_ws_resource_id(workspace_resource_id)
+
+    # extract subscription ID and resource group from workspace_resource_id URL
+    try:
+        subscription_id = workspace_resource_id.split('/')[2]
+        resource_group = workspace_resource_id.split('/')[4]
+        workspace_name = workspace_resource_id.split('/')[8]
+    except IndexError:
+        raise AzCLIError(
+            'Could not locate resource group in workspace-resource-id URL.')
+
+    # region of workspace can be different from region of RG so find the location of the workspace_resource_id
+    if not remove_monitoring:
+        resources = cf_resources(cmd.cli_ctx, subscription_id)
+        try:
+            resource = resources.get_by_id(
+                workspace_resource_id, '2015-11-01-preview')
+            location = resource.location
+        except (CloudError, HttpResponseError) as ex:
```

Comment:
We catch both CloudError and HttpResponseError. I suppose CloudError is used in track 1 SDK and HttpResponseError is used in track 2.

Reply:
Removing deprecated `CloudError` in commit [#32797f9](https://github.com/Azure/azure-cli/pull/20433/commits/32797f9645257b58b76080a3cc3fce1887fcada7).